### PR TITLE
ui: statements page search and pagination

### LINKED ISSUE #41537: ui: add search to statements page
Designs: https://zpl.io/bJ1JKGW

The current statements page in the UI can make it difficult to find a specific statement that you are looking for when troubleshooting performance. There are several reasons for this:

- The statements page currently only shows a summary of the statement fingerprint in the "Statement" column that includes the type of statement (SELECT, INSERT, UPDATE, etc.), and the table affected. The full statement is shown in a tooltip when the user hovers their mouse over the statement, or on the statement details page when a user clicks on a statement. This means that users cannot easily search the page for a specific statement using their browser's search function. 
- Because statements only show statement type and table, several statements will appear exactly the same in the statements column. The only way for a user to tell them apart today is to hover over each one or to click into each one.

To make this simpler for our users, we should add a search bar that allows the user to type to filter statements. The search should:

- Allow users to search the full statement fingerprint, not just the portion that is in the "Statements" column (cc @Annebirzin since we don't display the full statements currently, do we want to change that behavior when the user is typing to filter? what if they filter on a word that isn't in the column?)
- Allow users to type to filter, without requiring the user to hit enter to search (@elkmaster we should see how feasible this is. If this is lots of additional work over just searching when hitting enter, we can drop this requirements. Anne has mocked up designs for both possibilities)
- Allow users to search any part of the statement - users should be able to enter text from anywhere in the statement and get a list of all statement fingerprints that contain the entered text

### PR DESCRIPTION
**Pull request by vladlos**
Added search feature to statements page. Added paginated display (ui only).

Resolves: #41537, #41568

Release note (ui): statements page search and pagination
Screen Shot 2019-10-16 at 5 52 09 PM
Screenshot 2019-11-21 at 15 08 10

### CODE REVIEW

File: pkg/ui/src/views/app/components/Search/search.styl [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        border-right-width 2px !important
+  .ant-btn
+    margin 0
+    padding 0
+    width auto
+    height auto
+    border none
+    font-size 14px
+  ._prefix-icon
+    color #505565
```
[the remark this answers is not in the document]
Comment (celiala) [reply pkg/ui/src/views/app/components/Search/search.styl]: Update: before making variables within `.styl` file, see note below about implementing Design System palette.

File: pkg/ui/src/views/app/components/Search/search.styl [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      border-right-width 2px !important
+    &:placeholder
+      color $adminui-grey-2
+    &:not(:first-child)
+      padding-left 35px
+      padding-right 60px
+  ._submitted
+    .ant-input
+      &:not(:first-child)
+        padding-right 40px
```
Comment (celiala) [inline pkg/ui/src/views/app/components/Search/search.styl]: nit: add newline

File: pkg/ui/styl/shame.styl [hunk 3]
```diff
@@ -252,3 +275,109 @@
...
+    overflow-x hidde
+    overflow-y auto
+  .drawer__content
+    color #f6f6f6
+    font-family RobotoMono-Regular
+    font-size 12px
+    line-height 24px
+    ._text-bold
+      color #87beff
+      font-family RobotoMono-Bold
```
Comment (celiala) [inline pkg/ui/styl/shame.styl]: nit: add newline

**Review by celiala — APPROVED** [review]
Nice work! LGTM, modulo some super-minor nits :)

**Comment by nathanstilwell** [comment]
bors merge

**Comment by nathanstilwell** [comment]
bors merge

**Comment by nathanstilwell** [comment]
bors retry

**Comment by celiala** [comment]
bors retry

**Comment by celiala** [comment]
Aha -- @elkmaster , it looks like the build is failing because of this:
```
src/views/app/components/Search/index.tsx(75,11): error TS2605: JSX element type 'Input' is not a constructor function for JSX elements.
  The types returned by 'getSnapshotBeforeUpdate(...)' are incompatible between these types.
    Type 'null' is not assignable to type 'never'.
make: *** [ui-lint] Error 2
```

**Comment by vilterp** [comment]
Why is that failing in bors and passing on GitHub CI? 🤔

**Comment by nathanstilwell** [comment]
bors merge

**Pull request merged**